apisix/plugins/log-rotate.lua

```diff
@@ -219,18 +209,66 @@ local function init_default_logs(logs_info, log_type)
 end
 
 
+local function file_size(file)
+    local attr = lfs.attributes(file)
+    if attr then
+        return attr.size
+    end
+    return 0
+end
+
+
+local function rotate_file(files, now_time, max_kept)
+    for _, file in ipairs(files) do
+        local now_date = os_date("%Y-%m-%d_%H-%M-%S", now_time)
+        local new_file = rename_file(default_logs[file], now_date)
+        if not new_file then
+            return
+        end
+
+        local pid = process.get_master_pid()
+
+        core.log.warn("send USR1 signal to master process [", pid, "] for reopening log file")
+
+        if (pid) then
+            local ok, err = signal.kill(pid, signal.signum("USR1") or SIGUSR1)
+            if not ok then
+                core.log.error("failed to send USR1 signal for reopening log file: ", err)
+            end
+        end
+
+        if enable_compression then
+            compression_file(new_file)
+        end
+
+        -- clean the oldest file
+        local log_list, log_dir = scan_log_folder(file)
+        for i = max_kept + 1, #log_list do
+            local path = log_dir .. log_list[i]
+            local ok, err = os_remove(path)
+            if err then
+               core.log.error("remove old log file: ", path, " log: ", err, "  res:", ok)
+            end
+        end
+    end
+end
+
+
 local function rotate()
     local interval = INTERVAL
     local max_kept = MAX_KEPT
+    local max_size = -1
```

Comment:
Could you use a module-level constant like the others?

Reply:
done